### PR TITLE
Snappier lib

### PR DESCRIPTION
**Pull request by RichardSmedley**

### CODE REVIEW

File: modules/concept-docs/pages/compression.adoc [hunk 1]
```diff
@@ -8,11 +8,26 @@ include::project-docs:partial$attributes.adoc[]
 
 [abstract]
 In response to increasing volumes of data being sent over the wire, Couchbase Data Platform now provides data compression between the SDK and Couchbase Server.
-However, stable Snappy support is not yet available in .NET.
+However, stable Snappy support is not yet available in the .NET platform.
```
Comment (jeffrymorris) [inline modules/concept-docs/pages/compression.adoc:11]: This is no longer true, there is a couchbase approved Snappy lib [Snappier](https://github.com/brantburnett/Snappier) via https://github.com/couchbaselabs/Couchbase.Extensions/tree/master/src/Couchbase.Extensions.Compression.Snappier which is a plug-in into the SDK.

File: modules/concept-docs/pages/compression.adoc [hunk 2]
```diff
@@ -8,11 +8,26 @@ include::project-docs:partial$attributes.adoc[]
...
+
+
+== Community-Supported Snappy Library
+
+Whilst there is no official candidate for Snappy compression in the .NET platform, with several https://google.github.io/snappy/[unsupported choices available], the community has produced a promising option in https://github.com/brantburnett/Snappier[Snappier].
+And since xref:project-docs:sdk-release-notes.adoc#version-3-4-10[.NET SDK 3.4.10], the .NET SDK has been able to work with external Snappy libraries as the setting for the Server flag `SnappyEverywhere` is now supported.
+
+See the https://github.com/brantburnett/Snappier[Snappier GitHub page] for information on installing the external Snappier library, or get it from https://www.nuget.org/packages/Snappier[NuGet].
+
+CAUTION: Community-contributed Snappy libraries are not currently supported in production -- they are linked here for you to test out performance, in the run-up to an oficially-supported solution.
```
Comment (jeffrymorris) [inline modules/concept-docs/pages/compression.adoc]: Snappier has been approved for use by Couchbase as of 6/29/2023.

File: modules/concept-docs/pages/compression.adoc [hunk 3]
```diff
@@ -8,11 +8,26 @@ include::project-docs:partial$attributes.adoc[]
...
 
 Couchbase Server (in the Enterprise Edition) stores documents in compressed form, xref:7.1@server:learn:buckets-memory-and-storage/compression.adoc#compression-modes[when it is enabled in the server for a particular bucket], using Snappy Compression.
 As the Snappy compression library is not available for .NET, the server will automatically uncompress any compressed documents before sending them to the .NET client.
 
 If compression is set to _active_ on the server, documents will be stored there in compressed form, even though the .NET client has sent them uncompressed, thus saving storage space (but not network bandwidth).
+
+
+== Community-Supported Snappy Library
+
+Whilst there is no official candidate for Snappy compression in the .NET platform, with several https://google.github.io/snappy/[unsupported choices available], the community has produced a promising option in https://github.com/brantburnett/Snappier[Snappier].
```
Comment (jeffrymorris) [inline modules/concept-docs/pages/compression.adoc]: I wouldn't bring this up, just direct users to the Snappier library via the [Couchbase.Extensions](https://github.com/couchbaselabs/Couchbase.Extensions) [package](https://www.nuget.org/packages/Couchbase.Extensions.Compression.Snappier/1.0.0-beta.1). (Which makes me realize we need to push a 1.0.0 GA package.

File: modules/concept-docs/pages/compression.adoc [hunk 4]
```diff
@@ -8,11 +8,26 @@ include::project-docs:partial$attributes.adoc[]
...
+And since xref:project-docs:sdk-release-notes.adoc#version-3-4-10[.NET SDK 3.4.10], the .NET SDK has been able to work with external Snappy libraries as the setting for the Server flag `SnappyEverywhere` is now supported.
+
+See the https://github.com/brantburnett/Snappier[Snappier GitHub page] for information on installing the external Snappier library, or get it from https://www.nuget.org/packages/Snappier[NuGet].
```
Comment (jeffrymorris) [inline modules/concept-docs/pages/compression.adoc]: The integration with the SDK is via the [extensions](https://github.com/couchbaselabs/Couchbase.Extensions) plugin.

File: modules/concept-docs/pages/compression.adoc (hunk 1 above)
Comment (jeffrymorris) [inline modules/concept-docs/pages/compression.adoc]: I think we should link to the [.NET docs](https://dotnet.microsoft.com/) to make it clear was is meant here as I though you were referring to the SDK on first glance.

File: modules/concept-docs/pages/compression.adoc [hunk 5]
```diff
@@ -8,11 +8,26 @@ include::project-docs:partial$attributes.adoc[]
 
 [abstract]
 In response to increasing volumes of data being sent over the wire, Couchbase Data Platform now provides data compression between the SDK and Couchbase Server.
-However, stable Snappy support is not yet available in .NET.
+However, stable Snappy support is not yet available in the .NET platform.
 
 
 
 Couchbase Server (in the Enterprise Edition) stores documents in compressed form, xref:7.1@server:learn:buckets-memory-and-storage/compression.adoc#compression-modes[when it is enabled in the server for a particular bucket], using Snappy Compression.
 As the Snappy compression library is not available for .NET, the server will automatically uncompress any compressed documents before sending them to the .NET client.
```
Comment (jeffrymorris) [inline modules/concept-docs/pages/compression.adoc:17]: See my comments below.

**Review by jeffrymorris — CHANGES_REQUESTED** [review]
Hi Rchard -

I think we need to work on the messaging here as we do now have the "ok" to use Snappier in prod. I do think we need to define exactly what we actually support here however.

File: modules/concept-docs/pages/compression.adoc (hunk 5 above)
Comment (RichardSmedley) [reply modules/concept-docs/pages/compression.adoc:17]: Thanks @jeffrymorris - let me know if these changes cover it.

**Review by jeffrymorris — APPROVED** [review]

**Pull request merged**